ansible-scylla-node/tasks/generate_tokens.yml

```diff
@@ -16,38 +16,48 @@
   loop: "{{ wait_for_cql_port_output.results }}"
   when: bootstrapped_node is not defined and item.failed == False
 
-- name: Validate that all nodes are in UN state and save token lists
-  block:
-    - shell: |
-        nodetool ring | grep -E '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}'
-      register: _nodetool_ring_output
-    - set_fact:
-        token_list: "{{ token_list | default({}) | combine( {item.split()[0]: (token_list | default({}))[item.split()[0]] | default([]) + [item.split()[7]]} ) }}"
-      failed_when: item.split()[2] != 'Up' or item.split()[3] != 'Normal'
-      loop: "{{ _nodetool_ring_output.stdout_lines }}"
-  delegate_to: "{{ bootstrapped_node }}"
-  when: bootstrapped_node is defined
-
 - name: Prepare script arguments
   set_fact:
     node_list: "{{ node_list | default([]) + [hostvars[item]['broadcast_address']] }}"
     rack_list: "{{ rack_list | default([]) + [hostvars[item]['rack']] }}"
     dc_list: "{{ dc_list | default([]) + [hostvars[item]['dc']] }}"
   loop: "{{ groups['scylla'] }}"
 
-- name: Prepare old tokens list
-  set_fact:
-    token_list_str: "{{ token_list.keys() | zip(token_list.values()) | map('join', '=') | join('\n') | replace('[','') | replace(']','') | replace(\"'\", '') | replace(' ', '') }}"
-  when: token_list is defined
+- name: Create an empty tokens file
+  copy:
+    content: ""
+    dest: /tmp/tokens_file
+  delegate_to: localhost
```

Comment:
This won't work well when multiple concurrent instances of the job would be running.
Generate a temp file instead, please.

Reply:
Done